elastic/kibana
# Update roles management to handle creating, editing and listing custom roles in serverless

Closes https://github.com/elastic/kibana/issues/178913
Closes https://github.com/elastic/kibana/issues/178906
Closes https://github.com/elastic/kibana/issues/178973


## Summary

Updates the existing roles listing page to handle listing custom roles for serverless.
Updates the create/edit roles page to handle serverless custom roles. 

https://github.com/elastic/kibana/assets/1442420/10e2e779-2717-4802-932a-d9943a5cc937



## Testing

Following testing strategy laid out in: https://github.com/elastic/kibana/pull/176761 - set up as follows

1. In the `kibana.dev.yml` file, add the following settings. This enables the role management feature flag, and provides cloud URLs for the `Manage organization members` card.
```
xpack.security.roleManagementEnabled: true
xpack.cloud.base_url: 'https://cloud.elastic.co'
xpack.cloud.organization_url: '/account/members'
```
2. Start Elasticsearch and Kibana in serverless mode and SSL enabled (to access the test user selector) as follows:
```
yarn es --serverless=es --ssl -E xpack.security.authc.native_roles.enabled=true
yarn start --serverless=es --ssl
```
3. Navigate to Kibana (use https as SSL is enabled), and log in as the elastic_serverless user or as Admin test user. 
4. Navigate to the Management page using the side navigation bar. Verify the three new cards are rendered in a new Access section.
5. Using the custom roles card, navigate to the Custom Roles link. 
6. Verify the following
    a. The page shows the title and description as described above
    b. No reserved roles should be displayed
    c. An external link to the cloud console of your project to assign roles which navigates to the config above `xpack.cloud.base_url+xpack.cloud.organization_url`
7. Clicking on create role should take you to the create/edit roles page
8. Create a new role and assign it the privileges that you'd like
9. On clicking create role, you should be taken back to the role listing page and your new role should be displayed here along with a toast with a success message
10. On trying to delete the role, you will see a delete confirmation modal. On clicking delete, you should see the role removed from the role listing page along with a toast message.
